
This is a fix for the really ugly _hack_ to get the code compiled with recent
GCC's STLs, ...

Imported from FreeBSD via FreeNX-kNX to replace my own first quick hack fix.

  - Rene Rebe <rene@exactcode.de>

--- nxcomp/Types.h.orig	Wed Sep  1 21:18:18 2004
+++ nxcomp/Types.h	Wed Sep  1 21:21:41 2004
@@ -74,11 +74,11 @@
 
   void clear()
   {
-    #if defined(__STL_USE_STD_ALLOCATORS) || defined(__GLIBCPP_INTERNAL_VECTOR_H)
+    #if defined(__STL_USE_STD_ALLOCATORS) || defined(_VECTOR_H)
 
-    #if defined(__GLIBCPP_INTERNAL_VECTOR_H)
+    #if defined(_VECTOR_H)
 
-    _Destroy(_M_start, _M_finish);
+    _Destroy(_M_impl._M_start, _M_impl._M_finish);
 
     #else
 
@@ -86,9 +86,9 @@
 
     #endif
 
-    _M_deallocate(_M_start, _M_end_of_storage - _M_start);
+    _M_deallocate(_M_impl._M_start, _M_impl._M_end_of_storage - _M_impl._M_start);
 
-    _M_start = _M_finish = _M_end_of_storage = 0;
+    _M_impl._M_start = _M_impl._M_finish = _M_impl._M_end_of_storage = 0;
 
     #else
 
@@ -108,21 +108,21 @@
 
   void clear()
   {
-    #if defined(__STL_USE_STD_ALLOCATORS) || defined(__GLIBCPP_INTERNAL_VECTOR_H)
+    #if defined(__STL_USE_STD_ALLOCATORS) || defined(_VECTOR_H)
 
-    #if defined(__GLIBCPP_INTERNAL_VECTOR_H)
+    #if defined(_VECTOR_H)
 
-    _Destroy(_M_start, _M_finish);
+    _Destroy(_M_impl._M_start, _M_impl._M_finish);
 
     #else
 
-    destroy(_M_start, _M_finish);
+    destroy(_M_impl._M_start, _M_impl._M_finish);
 
     #endif
 
-    _M_deallocate(_M_start, _M_end_of_storage - _M_start);
+    _M_deallocate(_M_impl._M_start, _M_impl._M_end_of_storage - _M_impl._M_start);
 
-    _M_start = _M_finish = _M_end_of_storage = 0;
+    _M_impl._M_start = _M_impl._M_finish = _M_impl._M_end_of_storage = 0;
 
     #else
 
